src/librustc_trans/trans/datum.rs

```diff
@@ -529,7 +640,7 @@ impl<'tcx> Datum<'tcx, Lvalue> {
         };
         Datum {
             val: val,
-            kind: Lvalue,
+            kind: Lvalue::new("Datum::get_element"),
```

Comment:
Having post_store panic when it couldn't generate correct code was very helpful on my branch.  Maybe less useful here, but it might still be a good idea here to associate a "cannot be moved" hint with certain lvalues, like fields of lvalues with hints, which can't be moved safely.